### DISCUSSION TITLE
Is it still relevant to use varnish if Next.js can handle caching?

### DISCUSSION THREAD

**Discussion by helabenkhalfallah** [vercel/next.js#63525] [Help] [answered]
### Summary

We have an old stack that uses varnish to cache and boost performance. However, we are currently modernizing to transform CSR applications into SSR. As the SSR provides an HTML, it was said that it is possible to keep varnish to save the HTML. But reading the documentation of Next.js, we saw that it is possible to have a cache mechanism without using varnish. We wonder about the usefulness of varnish with the presence of cache capability with Next.js. What are the limits of Next.js cache vs varnish? 

Thank you very much for your help. :) :)

### Additional information
https://nextjs.org/docs/app/building-your-application/caching

**Comment by icyJoseph** [comment, marked as answer]
Hi, 

I think this page is better for your situation:

- https://nextjs.org/docs/app/building-your-application/deploying/production-checklist

>  Next.js caches data requests, the rendered result of Server and Client Components, static assets, and more, to reduce the number of network requests to your server, database, and backend services. You may opt out of caching, where appropriate.

Assets produced by the Next.js server, are have cache headers, that is JS/CSS/Fonts, etc, so that the browser won't request them as often.

And then in the Server, Next.js also caches some requests for 3rd party data. Then there's pre-rendering at build time, which makes it so that you can directly serve HTML for some pages, upon request, no need to do an SSR pass.

And the client is also caching React Server Component tree results, to reduce round trips.

Is there anything in particular that you wonder about? I remember that One.com uses Varnish, and that meant that no Cache Headers were applied, but that the Varnish server would make sure to delivery stuff fast... or something like that...

**Comment by helabenkhalfallah** [comment]
The idea is to keep the infrastructure light and efficient. If Next.js has caching capabilities, we said we can remove varnish from the chain (two things doing the cache). So we have question more about the limits of Next.js if we will remove varnish: scalability, speed, efficiency, duration, ... Thanks :)
